src/app/pages/data-protection/cloudsync/cloudsync-list/cloudsync-list.component.ts

```diff
@@ -140,27 +148,27 @@ export class CloudsyncListComponent implements EntityTableConfig<CloudSyncTaskUi
         icon: 'play_arrow',
         name: 'run',
         onClick: (row: CloudSyncTaskUi) => {
-          this.dialog
-            .confirm({ title: this.translate.instant('Run Now'), message: this.translate.instant('Run this cloud sync now?'), hideCheckBox: true })
-            .pipe(filter(Boolean), untilDestroyed(this)).subscribe(() => {
-              row.state = { state: JobState.Running };
-              this.ws.call('cloudsync.sync', [row.id]).pipe(untilDestroyed(this)).subscribe({
-                next: (jobId: number) => {
-                  this.dialog.info(
-                    this.translate.instant('Task Started'),
-                    this.translate.instant('Cloud sync <i>{taskName}</i> has started.', { taskName: row.description }),
-                    true,
-                  );
-                  this.job.getJobStatus(jobId).pipe(untilDestroyed(this)).subscribe((job: Job) => {
-                    row.state = { state: job.state };
-                    row.job = job;
-                  });
-                },
-                error: (err) => {
-                  new EntityUtils().handleWsError(this.entityList, err);
-                },
-              });
-            });
+          this.dialog.confirm({
+            title: this.translate.instant('Run Now'),
+            message: this.translate.instant('Run this cloud sync now?'),
+            hideCheckBox: true,
+          }).pipe(
+            filter(Boolean),
+            tap(() => row.state = { state: JobState.Running }),
+            switchMap(() => this.ws.call('cloudsync.sync', [row.id])),
+            tap(() => this.snackbar.success(
+              this.translate.instant('Cloud sync «{name}» has started.', { name: row.description }),
+            )),
+            switchMap((id) => this.store$.select(selectJob(id)).pipe(filter(Boolean))),
+            catchError((error) => {
+              this.dialog.errorReportMiddleware(error);
+              return EMPTY;
+            }),
+            untilDestroyed(this),
```

Comment:
I noticed that we have `untilDestroyed(this)` after all other pipes. I think it's better to put it as 1st pipe. This way it would prevent "zombie" API call and snackbar pop-up from a "dead" component.

Reply:
Yeah, that's intended.
<img width="681" alt="image" src="https://user-images.githubusercontent.com/351613/199422928-b29a44c1-ea16-4ad4-aceb-ad4dd60a34f5.png">